File: tests/csharp/tests-csharp-nunit-allure3-restsharp/Api/HttpResult.cs
using System.Text.Json;

namespace Api;

public sealed record HttpResult(int Status, string Body)
{
    private JsonElement Json
    {
        get
        {
            using var doc = JsonDocument.Parse(string.IsNullOrWhiteSpace(Body) ? "{}" : Body);
            return doc.RootElement.Clone();
        }
    }

    public string Text(string field) =>
        Json.TryGetProperty(field, out var value) ? value.GetString() ?? "" : "";

    public IReadOnlyList<string> ItemNames()
    {
        if (!Json.TryGetProperty("items", out var items) || items.ValueKind != JsonValueKind.Array)
        {
            return [];
        }

        return items.EnumerateArray()
            .Select(item => item.TryGetProperty("name", out var name) ? name.GetString() ?? "" : "")
            .ToList();
    }
}

File: tests/java/tests-java-junit5-api_request-playwright/src/test/java/api/HttpResult.java
package api;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.util.ArrayList;
import java.util.List;

public record HttpResult(int status, String body) {

    static final ObjectMapper MAPPER = new ObjectMapper();

    public JsonNode json() {
        try {
            return MAPPER.readTree(body == null || body.isBlank() ? "{}" : body);
        } catch (Exception e) {
            throw new IllegalStateException("not JSON: " + body, e);
        }
    }

    public String text(String field) {
        return json().path(field).asText();
    }

    public List<String> itemNames() {
        List<String> names = new ArrayList<>();
        for (JsonNode item : json().path("items")) {
            names.add(item.path("name").asText());
        }
        return names;
    }
}
